cargo/cargo_toml_info/main.rs

```diff
@@ -0,0 +1,272 @@
+//! Command line tool invoked by Bazel to read metadata out of a `Cargo.toml` file.
+//!
+//! This tool should _not_ be used to determine dependencies, features, or generally any build
+//! information for a crate, that should live in crate_universe. This tool is intended to read
+//! non-build related Cargo metadata like lints, authors, or badges.
+
+use cargo_toml::{Lint, LintLevel, Manifest};
+
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::error::Error;
+use std::fs::File;
+use std::io::{LineWriter, Write};
+use std::path::PathBuf;
+
+pub fn main() -> Result<(), Box<dyn Error>> {
+    let Args {
+        manifest_toml,
+        workspace_toml,
+        command,
+    } = Args::try_from(std::env::args())?;
+
+    let manifest_contents = std::fs::read_to_string(&manifest_toml)?;
+    let mut crate_manifest = Manifest::from_str(&manifest_contents)?;
+    let mut workspace_manifest = None;
+
+    // Optionally populate the manifest with info from the parent workspace, if one is provided.
+    if let Some(workspace_path) = workspace_toml {
+        let manifest = Manifest::from_path(&workspace_path)?;
+        let workspace_details = Some((&manifest, workspace_path.as_path()));
+
+        // TODO(parkmycar): Fix cargo_toml so we inherit lints from our workspace.
+        crate_manifest.complete_from_path_and_workspace(&manifest_toml, workspace_details)?;
+        workspace_manifest = Some(manifest);
+    }
+
+    match command {
+        Command::Lints(args) => {
+            generate_lints_info(&crate_manifest, workspace_manifest.as_ref(), args)?
+        }
+    }
+
+    Ok(())
+}
+
+#[derive(Debug)]
+struct LintsArgs {
+    output_rustc_lints: PathBuf,
+    output_clippy_lints: PathBuf,
+    output_rustdoc_lints: PathBuf,
+    // TODO(parkmycar): Support Rust's new --check-cfg once cargo-toml is updated.
```

Comment:
Can we link to an issue?

Reply:
Filed and linked to [cargo_toml#34](https://gitlab.com/lib.rs/cargo_toml/-/issues/34)!